cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableLogInspectCommand.java

```diff
@@ -0,0 +1,334 @@
+                                " \"StorageMetadataCheckpointOperation|StreamSegmentAppendOperation|StreamSegmentMapOperation|\" +\n" +
+                                " \"StreamSegmentSealOperation|StreamSegmentTruncateOperation|UpdateAttributesOperation]");
+                        predicates.add(a -> a.getOperationTypeString().equals(op));
+                        break;
+                    case SEQUENCE_NUMBER:
+                    case SEGMENT_ID:
+                    case OFFSET:
+                    case LENGTH:
+                    case ATTRIBUTES:
+                        op = getStringUserInput("Search operation based on: [value/range]");
+                        if (op.equals("value")) {
+                            long in = getLongUserInput("Enter valid " + conditionTpe);
+                            predicates.add(a -> a.getSequenceNumber() == in);
+                        } else {
+                            predicates.add(valueOrRangeInput(conditionTpe));
+                        }
+                        break;
+                    default:
+                        showMessage = false;
+                        output("Invalid operation, please select one of [OperationType/SequenceNumber/SegmentId/Offset/Length/Attributes]");
+                }
+                predicate = clause.equals(OPERATOR_AND) ?
+                        predicates.stream().reduce(Predicate::and).orElse(x -> true) : predicates.stream().reduce(Predicate::or).orElse(x -> false);
+            } catch (NumberFormatException ex) {
+                outputError("Wrong input argument.");
+                outputException(ex);
+            } catch (Exception ex) {
+                outputError("Some problem has happened.");
+                outputException(ex);
+            }
+            if (showMessage) {
+                output("You can continue adding conditions for inspect.");
+            }
+            finishInputCommands = !confirmContinue();
+            next = !finishInputCommands;
+        }
+        output("Value of predicates is : " + predicates);
+        return predicate;
+    }
+
+    /**
+     * Guides the users to a set of options for creating range for selected
+     * option by the user.
+     *
+     * @return Predicate<OperationInspectInfo>.
+     */
+    private Predicate<OperationInspectInfo> valueOrRangeInput(String conditionTpe) {
+        List<Predicate<OperationInspectInfo>> predicates = new ArrayList<>();
+        Predicate<OperationInspectInfo> predicate = null;
+        String clause = OPERATOR_AND;
+        boolean finish = false, next = false;
+        while (!finish) {
+            if (next) {
+                clause = getStringUserInput("Select conditional operator: [and/or]");
+            }
+            final String input = getStringUserInput("Select operator : [</>/<=/>=/!=]");
+            switch (input) {
+                case LESS_THAN:
+                    long in = getLongUserInput("Enter range " + conditionTpe);
```

Comment:
Could this go outside the `switch` block? It seems to be repeated and perhaps it could be avoided.